src/platform/plugins/shared/workflows_management/public/features/run_workflow/ui/workflow_execute_index_form.tsx

```diff
@@ -0,0 +1,407 @@
+        render: (timestamp: string) => (
+          <EuiText size="s">{new Date(timestamp).toLocaleString()}</EuiText>
+        ),
+      },
+      {
+        field: '_source',
+        name: 'Document',
+        render: (source: any) => {
+          const listItems: Array<{ title: string; description: string }> = [];
+          ['kind', 'agent', 'user', 'message'].forEach((field: string) => {
+            if (source[field] === undefined) {
+              return;
+            }
+            listItems.push({
+              title: field,
+              description: source[field] || '-',
+            });
+          });
+          return (
+            <EuiFlexGroup alignItems="center" gutterSize="s">
+              <EuiFlexItem grow={false}>
+                <EuiToken iconType="tokenString" />
+              </EuiFlexItem>
+              <EuiFlexItem>
+                <EuiDescriptionList type="inline" listItems={listItems} />
+              </EuiFlexItem>
+            </EuiFlexGroup>
+          );
+        },
+      },
+    ],
+    []
+  );
+
+  // Table selection configuration
+  const selection = {
+    onSelectionChange: handleDocumentSelection,
+    selectable: (item: Document) => true,
+    selectableMessage: (selectable: boolean) => (!selectable ? 'Document not selectable' : ''),
+  };
+
+  return (
+    <EuiFlexGroup direction="column" gutterSize="l">
+      <EuiSpacer size="s" />
+      <EuiFlexGroup direction="row" gutterSize="l">
+        {/* Data View Selector */}
+        <EuiFlexItem
+          grow={false}
+          css={css`
+            width: 200px;
+          `}
+        >
+          <EuiFormRow label="Data View">
+            <EuiComboBox
+              placeholder="Select data view"
+              aria-label="Data view"
+              options={dataViewOptions}
+              selectedOptions={
+                selectedDataView
+                  ? dataViewOptions.filter((opt) => opt.value === selectedDataView.id)
+                  : []
+              }
+              onChange={(selectedOptions) => {
+                if (selectedOptions.length > 0) {
+                  handleDataViewChange(selectedOptions[0].value as string);
+                }
+              }}
+              singleSelection={{ asPlainText: true }}
+              isClearable={false}
+              data-test-subj="workflow-data-view-selector"
+            />
+          </EuiFormRow>
+        </EuiFlexItem>
+        <EuiFlexItem>
+          <EuiFormRow label="Documents" fullWidth>
+            <EuiFieldText
+              placeholder="Filter your data using KQL syntax"
+              value={queryString}
+              onChange={(e) => setQueryString(e.target.value)}
+              onKeyPress={(e) => {
+                if (e.key === 'Enter') {
+                  fetchDocuments();
+                }
+              }}
+              fullWidth
+              prepend={<EuiIcon type="search" />}
+              data-test-subj="workflow-query-input"
+            />
```

Comment:
For the search bar, I think it would be nice to try reusing the `SearchBar` component. 

To get it:

```
const { unifiedSearch } = useKibana().services.unifiedSearch;
const { SearchBar } = unifiedSearch.ui;
```

Example here:
https://github.com/elastic/kibana/blob/main/x-pack/solutions/security/plugins/security_solution/public/common/components/search_bar/index.tsx

<img width="1384" height="119" alt="Captura de pantalla 2025-09-26 a les 13 11 25" src="https://github.com/user-attachments/assets/ff559a74-61f8-4462-8588-eca5dfd19d76" />

And maybe it would make sense to add a timerange input as well to the form (maybe later)